src/asset-swapper/types.ts

```diff
@@ -33,16 +33,26 @@ export interface OrderPrunerOpts {
     permittedOrderFeeTypes: Set<OrderPrunerPermittedFeeTypes>;
 }
 
-export interface SignedOrder<T> {
-    order: T;
-    type: FillQuoteTransformerOrderType.Limit | FillQuoteTransformerOrderType.Rfq | FillQuoteTransformerOrderType.Otc;
+export interface SignedOrder {
     signature: Signature;
 }
 
-export type SignedNativeOrder =
-    | SignedOrder<LimitOrderFields>
-    | SignedOrder<RfqOrderFields>
-    | SignedOrder<OtcOrderFields>;
+export interface SignedLimitOrder extends SignedOrder {
```

Comment:
I think it's a little bit clearer if you just drop the`extends` and put `signature` in each of these interfaces

Reply:
Sounds good! 
I was actually debating between two lol